### PR TITLE
Make dashboard filter sticky

### LINKED ISSUE #11187: Dashboard filters should be pinned on top / sticky on top
**Is your feature request related to a problem? Please describe.**
We’ve implemented a few dashboards on metabase, and noticed an important UX issues for dashboards that have more than a few charts: the filters disappear as soon as you scroll down.

Now clearly that’s not a bug, and that’s the expected behaviour for metabase dashboard.

Two workflows aren’t possible when filters are not visible:

Checking that the correct filter is selected (quite important when presenting)
Exploring data (right now you have to scroll up, change the filter, then scroll down to see changes - not exactly ideal)

**Describe the solution you'd like**
Having a sticky bar with the filters on top (or some other design) would allow these two workflows.
Or designs are fine, what really matters is that the user can always see and interact with the filters

**Describe alternatives you've considered**
N/A
**How important is this feature to you?**
Makes it hard to use metabase for dashboards that don't fit on a single screen, which is quite limiting

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### PR DESCRIPTION
**Pull request by gusaiani**
Fixes #11187

## How to Test

1. Visit dashboards that are tall enough to make the window scrollable.
2. Scroll
3. Filter should now be sticky at the top on scroll.

### Demo

Metabase Scroll 1

### CODE REVIEW

**Review by alxnddr — COMMENTED** [review]
Looks promising on the dashboards inside the app! 
I just found a couple of issues you might have missed:

Add question sidebar layout is not correct
Screen Shot 2021-08-04 at 22 41 57

When dashboards are publicly shared, the filter panel is not sticky.
Screen Shot 2021-08-04 at 22 44 51

How it should behave on embedded dashboards inside iframes?

File: frontend/src/metabase/css/core/layout.css [hunk 1]
```diff
@@ -5,6 +5,7 @@
   --xl-width: 1540px;
 }
 
+// When converting to styled components, use FullWidthContainer
```
Comment (alxnddr) [inline frontend/src/metabase/css/core/layout.css:8]: `//` is not a valid comment in css, it probably should be `/* When converting to styled components, use FullWidthContainer */`

**Comment by gusaiani** [comment]
@alxnddr Thank you very much for the review.
Yes, a simple UI check I should have done :)

In you opinion, would it be good add a few visual regression tests to these states of the dashboard in this PR?
To me it looks so.
If it does for you, any ideas off the top of your head on how to approach it, even in a separate PR?

**Comment by alxnddr** [comment]
@gusaiani that would be fantastic to add some visual tests on dashboards! There is an example test for scatter which you can use as an example or just reach out to me for any help. I'm not sure about visual tests nicely capturing fixed/sticky elements, we need to check that

**Comment by nemanjaglumac** [comment]
> I'm not sure about visual tests nicely capturing fixed/sticky elements, we need to check that

We can scroll the page down and then take a screenshot. It should expect to see the filter there.

**Review by flamber — COMMENTED** [review]
Some quick testing.
1. Is this only supposed to make the filters sticky, when viewing in-app dashboard?
A lot of people are having the problem, while editing a larger dashboard, where they have to scroll up and down a lot.
2. There's a horizontal scroll
image
3. Some sidebars are off-screen
image
image
4. The filter sidebar is on-screen, but it's not being positioned anymore, so it scrolls with body
image
5. If a filter has a dropdown, then it scrolls to top

https://user-images.githubusercontent.com/1447303/128399772-a63c0071-8ea9-411b-a96d-17b4ce576b97.mp4

**Comment by gusaiani** [comment]
Thank you @flamber.

A clean fix like we have here now is looking unlikely to work (position: sticky depends on a lot of things we are not doing in ancestor elements).

> Is this only supposed to make the filters sticky, when viewing in-app dashboard?

By this you mean dashboards that are not shared publicly?

If that is what you mean, yes, the idea is for embedded or shared dashboards to have them sticky too.
I've been researching this a bit. At the very least, looks like a separate PR later on.

**Comment by flamber** [comment]
No, I'm asking if filters will be sticky in edit-mode. Try creating a dashboard with 8 filter and 30 cards, and then try to connect the filters - then you'll understand ;-)

**Comment by gusaiani** [comment]
@flamber 

> No, I'm asking if filters will be sticky in edit-mode. Try creating a dashboard with 8 filter and 30 cards, and then try to connect the filters - then you'll understand ;-)

[From the "spec"](https://www.notion.so/metabase/Make-dashboard-filters-sticky-4f888413a6d0413dbd5eedaba4b9e8d5#f7fff53b38774f0288de0b1135d44d0c), it wasn't a firm requirement either way.

Initially I made it sticky for both viewing and editing, only later I restricted it to viewing. I'm not aware of how many more side-effects it would introduce, but at first sight it looked not a lot more complicated.

Will keep this in mind and bring it to up to product.

**Comment by gusaiani** [comment]
Ok, learned a bunch of things about the Dashboard.

Now...Let's improve the Dashboard component file first.

**Pull request closed without merging**